### PR TITLE
feat(platform): auto-discover library route directories and fail build on route collisions

### PR DESCRIPTION
**Pull request by benpsnyder**
## PR Checklist

Follow-up to [Split content routing into @analogjs/router/content analogjs/analog#2216](https://github.com/analogjs/analog/pull/2216) — addresses maintainer feedback that routes/content/api defined in `libs/shared/feature` should be picked up without explicit `additional*Dirs` configuration, and hardens the build against ambiguous route definitions.

Closes [analogjs/analog#2202](https://github.com/analogjs/analog/issues/2202) (follow-up)

## Affected scope

- Primary scope: platform
- Secondary scopes: router (collision detection contracts)

## Recommended merge strategy for maintainer

- [x] Squash merge
- [ ] Rebase merge
- [ ] Other

### 1. Auto-discover library route directories (`discoverRoutes: true`)

A new opt-in option reads `tsconfig.base.json` path aliases and checks each library for conventional route directories:

| Detected directory | Maps to option | Path format |
|:--|:--|:--|
| `libs/<name>/src/pages/` | `additionalPagesDirs` | `/<libRoot>` |
| `libs/<name>/src/content/` | `additionalContentDirs` | `/<libRoot>/src/content` |
| `libs/<name>/src/api/` | `additionalAPIDirs` | `/<libRoot>/src/api` |

Before:
```ts
analog({
  additionalPagesDirs: ['/libs/shared/feature'],
  additionalContentDirs: ['/libs/shared/feature/src/content'],
  additionalAPIDirs: ['/libs/shared/feature/src/api'],
})
```

After:
```ts
analog({
  discoverRoutes: true,
})
```

- Skips `@analogjs/*` framework packages — only scans `libs/` entries
- Falls back to `tsconfig.json` when `tsconfig.base.json` is missing
- Merges with explicit `additional*Dirs` (user config takes precedence, deduped)
- Backward compatible: defaults to `false`

### 2. Build-time error on same-priority route collisions

When two source files at the same priority level resolve to the same route path, the build now fails with a clear error:

```
[analog] Route collisions detected during build:
  '/test': '/src/app/pages/test.page.ts' vs '/src/content/test.md'

Each route path must be defined by exactly one source file.
Remove or rename the conflicting files to resolve the collision.
```

### 3. Smart collision exclusions

Three categories of same-path routes are intentional and are excluded from collision detection:

| Pattern | Example | Why it's valid |
|:--|:--|:--|
| **Group routes** | `(auth).page.ts` + `(home).page.ts` | Pathless layout wrappers that coexist at the same URL |
| **Layout + index** | `docs.page.ts` + `docs/index.page.ts` | Parent layout wraps child index — standard Analog routing |
| **Duplicate discovery** | Same file surfaced twice | Dedup from overlapping globs |

Cross-priority collisions (app-local overriding a library route) remain warnings only, since that precedence is intentional.

### 4. `RouteManifest` collision tracking

`RouteManifest` now includes a `collisions` array so downstream consumers can inspect detected conflicts:

```ts
interface RouteCollision {
  fullPath: string;
  keptFile: string;
  droppedFile: string;
  samePriority: boolean; // true = hard error, false = precedence warning
}
```

## Test plan

- [x] `nx test platform` — 204 tests pass (14 files)
  - 10 new `discover-library-routes.spec.ts` tests (tsconfig parsing, filtering, edge cases)
  - 3 new `platform-plugin.spec.ts` integration tests (merging, dedup, opt-in guard)
  - 4 new `route-manifest.spec.ts` tests (collision tracking, `samePriority` flag, group-route exclusion, layout+index exclusion, duplicate-file skip)
  - 1 new `typed-routes-plugin.spec.ts` test (build error on same-priority collision)
- [x] `nx run analog-app:build:production` — 19 routes, builds successfully
- [x] `nx run opt-catchall-app:build:production` — builds successfully (layout+index pair at `/docs` correctly skipped)
- [ ] `nx format:check`
- [ ] `pnpm build`
- [ ] `pnpm test`

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

`discoverRoutes` defaults to `false`. The build-time collision error only fires for genuine same-priority conflicts that were previously silently dropping routes — surfacing a latent bug rather than changing correct behavior.

## Other information

**Files changed:**

| File | Change |
|:--|:--|
| `packages/platform/src/lib/discover-library-routes.ts` | New: reads tsconfig paths, filters to `libs/`, checks `existsSync` for conventional subdirs |
| `packages/platform/src/lib/discover-library-routes.spec.ts` | New: 10 unit tests |
| `packages/platform/src/lib/options.ts` | Added `discoverRoutes?: boolean` |
| `packages/platform/src/lib/platform-plugin.ts` | Wires discovery into option enrichment |
| `packages/platform/src/lib/platform-plugin.spec.ts` | 3 integration tests |
| `packages/platform/src/lib/route-manifest.ts` | `RouteManifest.collisions` array with `samePriority` flag, group-route exclusion, layout+index exclusion, duplicate-file skip |
| `packages/platform/src/lib/route-manifest.spec.ts` | 4 collision tests |
| `packages/platform/src/lib/typed-routes-plugin.ts` | Build error on same-priority collisions |
| `packages/platform/src/lib/typed-routes-plugin.spec.ts` | Build error test |
| `apps/analog-app/vite.config.ts` | `discoverRoutes: true` replaces manual dirs |
| `apps/blog-app/vite.config.ts` | `discoverRoutes: true` replaces manual dirs |
| `libs/shared/feature/src/content/shared-test.md` | Renamed from `test.md` to resolve `/test` collision with `test.page.ts` |
| `libs/shared/feature/src/content/another-test.md` | New content file for testing |

### CODE REVIEW

**Comment by benpsnyder** [comment]
### Build-time collision detection

The deploy preview caught two pre-existing route collisions that were previously silent warnings:

1. **`/` — `(auth).page.ts` vs `(home).page.ts`**: False positive — both are group/layout routes that intentionally share the same URL path. Fixed by skipping collision recording when both files are group routes.

2. **`/test` — `test.md` vs `test.page.ts`**: Genuine collision in `libs/shared/feature/src/`. Fixed by renaming `test.md` → `shared-test.md`.

The collision guard now distinguishes:
| Collision type | Dev mode | Build mode |
|:--|:--|:--|
| Same priority (e.g., page + content at same path) | Warning | **Error** |
| Cross priority (app-local overrides library route) | Warning | Warning |
| Group routes sharing a path (e.g., `(auth)` + `(home)`) | Skipped | Skipped |

image

File: packages/platform/src/lib/route-manifest.spec.ts [hunk 1]
```diff
@@ -300,6 +301,25 @@ describe('generateRouteManifest', () => {
...
+      '/src/app/pages/about.page.ts',
+      '/src/app/pages/about.page.ts',
+    ]);
+
+    expect(manifest.routes).toHaveLength(1);
+    expect(manifest.collisions).toHaveLength(0);
+    expect(spy).not.toHaveBeenCalled();
 
     spy.mockRestore();
   });
```
[the remark this answers is not in the document]
Comment (benpsnyder) [reply packages/platform/src/lib/route-manifest.spec.ts:356]: should be addressed

**Comment by benpsnyder** [comment]
@brandonroberts LGTM
